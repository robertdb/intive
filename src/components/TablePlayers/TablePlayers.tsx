import * as React from 'react';
import "react-super-responsive-table/dist/SuperResponsiveTableStyle.css"
import {
    WrapperTable
} from './style';

declare var require: (moduleId: string) => any;
let TableResponsive = require("react-super-responsive-table");
const { Table, Thead, Tbody, Tr, Th, Td } = TableResponsive;

class TablePlayers extends React.Component {
    public render() {
        return (
            <WrapperTable>
                <Table>
                    <Thead>
                        <Tr>
                            <Th>Player</Th>
                            <Th>Position</Th>
                            <Th>Team</Th>
                            <Th>Age</Th>
                        </Tr>
                    </Thead>
                    <Tbody>
                        <Tr>
                            <Td>Tablescon</Td>
                            <Td>9 April 2019</Td>
                            <Td>East Annex</Td>
                            <Td>Eaa22st Annex</Td>
                        </Tr>
                        <Tr>
                            <Td>Capstone Data</Td>
                            <Td>19 May 2019</Td>
                            <Td>205 Gorgas</Td>
                            <Td>Easafdsft Annex</Td>
                        </Tr>
                        <Tr>
                            <Td>Tuscaloosa D3</Td>
                            <Td>29 June 2019</Td>
                            <Td>Github</Td>
                            <Td>Gitasdsdhub</Td>
                        </Tr>
                    </Tbody>
                </Table>
            </WrapperTable>
        );
    }
}

export default TablePlayers;
