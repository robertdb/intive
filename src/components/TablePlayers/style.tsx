import styled from 'styled-components';

export const WrapperTable = styled.div`
    margin-top: 2rem;
    & > table > thead > tr > th {
        background-color: #000;
        color: #fff;
        font-size: 1.5rem;
    }
    & > table > tbody > tr > td {
        padding: 20px
        font-size: 1rem;
        background-color: #ccc;
    }
`;
