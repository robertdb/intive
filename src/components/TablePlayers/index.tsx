import TablePlayers from './TablePlayers';

export default TablePlayers;