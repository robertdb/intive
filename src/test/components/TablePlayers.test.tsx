import * as React from 'react';
import * as enzyme from 'enzyme';
import '../setupTest';
import TablePlayers from '../../components/TablePlayers';

const { mount, shallow } = enzyme;

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<TablePlayers players={[]} />);

        expect(component).toMatchSnapshot();
    });
});

describe('Basic Table rules', () => {
    it('There should be ONLY 1 table element', () => {
        const wrapper = mount(<TablePlayers players={[]} />);
        // There should be ONLY 1 table element
        const table = wrapper.find('table');
        expect(table).toHaveLength(1);
    });
    it('The table should have ONLY 1 thead element', () => {
        const wrapper = mount(<TablePlayers players={[]} />);
        const table = wrapper.find('table');
        const thead = table.find('thead');
        expect(thead).toHaveLength(1);
    });
    it('The table should have ONLY 1 tbody tag', () => {
        const wrapper = mount(<TablePlayers players={[]} />);
        const table = wrapper.find('table');
        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);
    });
});

describe('Table player should render data always', () => {
    it('There should be ONLY 1 Tr in tbody element with empty array', () => {
        const wrapper = mount(<TablePlayers players={[]} />);
        const table = wrapper.find('table');
        const tbody = table.find('tbody');
        const tr = tbody.find('tr');
        expect(tr).toHaveLength(1);
    });
    it('There should be ONLY 1 Tr in tbody element with empty array', () => {
        const data = [{ name: 'messi', position: "striker", nationality: "argentine", age: "32" }]
        const wrapper = mount(<TablePlayers players={data} />);
        const table = wrapper.find('table');
        const tbody = table.find('tbody');
        const tr = tbody.find('tr');
        expect(tr).toHaveLength(1);
    });
});

