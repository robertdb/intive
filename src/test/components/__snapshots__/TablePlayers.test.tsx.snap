// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TablePlayers
    players={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Table>
        <Thead>
                <Tr>
                        <Th>
                                Player
                        </Th>
                        <Th>
                                Position
                        </Th>
                        <Th>
                                Nationality
                        </Th>
                        <Th>
                                Age
                        </Th>
                </Tr>
        </Thead>
        <Tbody>
                <Tr>
                        <Td>
                                -
                        </Td>
                        <Td>
                                -
                        </Td>
                        <Td>
                                -
                        </Td>
                        <Td>
                                -
                        </Td>
                </Tr>
        </Tbody>
</Table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Thead>
            <Tr>
                        <Th>
                                    Player
                        </Th>
                        <Th>
                                    Position
                        </Th>
                        <Th>
                                    Nationality
                        </Th>
                        <Th>
                                    Age
                        </Th>
            </Tr>
</Thead>,
          <Tbody>
            <Tr>
                        <Td>
                                    -
                        </Td>
                        <Td>
                                    -
                        </Td>
                        <Td>
                                    -
                        </Td>
                        <Td>
                                    -
                        </Td>
            </Tr>
</Tbody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Tr>
              <Th>
                            Player
              </Th>
              <Th>
                            Position
              </Th>
              <Th>
                            Nationality
              </Th>
              <Th>
                            Age
              </Th>
</Tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Th>
                  Player
</Th>,
                <Th>
                  Position
</Th>,
                <Th>
                  Nationality
</Th>,
                <Th>
                  Age
</Th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Player",
                },
                "ref": null,
                "rendered": "Player",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Position",
                },
                "ref": null,
                "rendered": "Position",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Nationality",
                },
                "ref": null,
                "rendered": "Nationality",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Age",
                },
                "ref": null,
                "rendered": "Age",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Tr>
                <Td>
                                -
                </Td>
                <Td>
                                -
                </Td>
                <Td>
                                -
                </Td>
                <Td>
                                -
                </Td>
</Tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Td>
                    -
</Td>,
                  <Td>
                    -
</Td>,
                  <Td>
                    -
</Td>,
                  <Td>
                    -
</Td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    margin-top: 2rem;
    & > table > thead > tr > th {
        background-color: #000;
        color: #fff;
        font-size: 1.5rem;
    }
    & > table > tbody > tr > td {
        padding: 20px
        font-size: 1rem;
        background-color: #ededed;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Table>
          <Thead>
                    <Tr>
                              <Th>
                                        Player
                              </Th>
                              <Th>
                                        Position
                              </Th>
                              <Th>
                                        Nationality
                              </Th>
                              <Th>
                                        Age
                              </Th>
                    </Tr>
          </Thead>
          <Tbody>
                    <Tr>
                              <Td>
                                        -
                              </Td>
                              <Td>
                                        -
                              </Td>
                              <Td>
                                        -
                              </Td>
                              <Td>
                                        -
                              </Td>
                    </Tr>
          </Tbody>
</Table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Thead>
              <Tr>
                            <Th>
                                          Player
                            </Th>
                            <Th>
                                          Position
                            </Th>
                            <Th>
                                          Nationality
                            </Th>
                            <Th>
                                          Age
                            </Th>
              </Tr>
</Thead>,
            <Tbody>
              <Tr>
                            <Td>
                                          -
                            </Td>
                            <Td>
                                          -
                            </Td>
                            <Td>
                                          -
                            </Td>
                            <Td>
                                          -
                            </Td>
              </Tr>
</Tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Tr>
                <Th>
                                Player
                </Th>
                <Th>
                                Position
                </Th>
                <Th>
                                Nationality
                </Th>
                <Th>
                                Age
                </Th>
</Tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Th>
                    Player
</Th>,
                  <Th>
                    Position
</Th>,
                  <Th>
                    Nationality
</Th>,
                  <Th>
                    Age
</Th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Player",
                  },
                  "ref": null,
                  "rendered": "Player",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Position",
                  },
                  "ref": null,
                  "rendered": "Position",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Nationality",
                  },
                  "ref": null,
                  "rendered": "Nationality",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Age",
                  },
                  "ref": null,
                  "rendered": "Age",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Tr>
                  <Td>
                                    -
                  </Td>
                  <Td>
                                    -
                  </Td>
                  <Td>
                                    -
                  </Td>
                  <Td>
                                    -
                  </Td>
</Tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Td>
                      -
</Td>,
                    <Td>
                      -
</Td>,
                    <Td>
                      -
</Td>,
                    <Td>
                      -
</Td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    margin-top: 2rem;
    & > table > thead > tr > th {
        background-color: #000;
        color: #fff;
        font-size: 1.5rem;
    }
    & > table > tbody > tr > td {
        padding: 20px
        font-size: 1rem;
        background-color: #ededed;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
